src/Sylius/Bundle/CoreBundle/CatalogPromotion/Processor/CatalogPromotionArchivalProcessorInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\CatalogPromotion\Processor;
+
+interface CatalogPromotionArchivalProcessorInterface
+{
+    public function archiveCatalogPromotion(string $catalogPromotionCode): void;
```

Comment:
I see long method names are in line with other processors and using `$code` instead of `CatalogPromotionInterface` is similar to  `CatalogPromotionRemovalProcessor`.

but I personally would have preferred something like this

```suggestion
    public function archive(CatalogPromotionInterface  $catalogPromotion): void;
    public function archiveByCode(string $catalogPromotionCode): void;
```

Reply:
I agree with you. But I use the processor directly in controller and wanted to avoid any logic there to load the promotion from repository, assert it etc. And still, it's consistent with other catalog promotion processor, so I would let this as is, if you don't mind :vulcan_salute: